cmd/ctr/commands/containers/containers.go

```diff
@@ -194,16 +196,31 @@ func deleteContainer(ctx context.Context, client *containerd.Client, id string,
 	}
 	status, err := task.Status(ctx)
 	if err != nil {
+		if isNotFoundError(err) {
+			return container.Delete(ctx, opts...)
```

Comment:
How about doing that in defer?

Reply:
The `container.Delete` may also return error, we can't distinguish the cause, I think It's a little hard to do that in defer